src/context/theme-context.jsx

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useState } from "react";
+
+const ThemeContext = createContext(null);
+
+const useTheme = () => useContext(ThemeContext);
+
+const ThemeProvider = ({ children }) => {
+  // fetches user system theme preference
+  const prefersDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
```

Comment:
wouldn't this fetch the preferred color scheme each time?

No way of remembering if the user changes the theme for this particular app?

Reply:
Yes it will fetch the system theme every render.
It can be stored in localstorage to maintain consistency, 
Good catch, thanks for pointing it out!